File: solutions/1887. Reduction Operations to Make the Array Elements Equal/1887.cpp
// Time complexity: O(sort)
// Space complexity: O(sort)
class Solution {
public:
    int reductionOperations(vector<int>& nums) {
        ranges::sort(nums);
        int ans = 0;
        int up = 0;
        
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] != nums[i - 1]) {
                ++up;
            }
            
            ans += up;
        }
        
        return ans;
    }
};

File: solutions/1887. Reduction Operations to Make the Array Elements Equal/1887.rs
// Time complexity: O(sort)
// Space complexity: O(sort)
impl Solution {
    pub fn reduction_operations(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut ans = 0;
        let mut up = 0;

        for i in 1..nums.len() {
            if nums[i] != nums[i - 1] {
                up += 1;
            }
            ans += up;
        }
        return ans;
    }
}